modules/olm-disconnected-operatorhub-disconnected-mode.adoc

```diff
@@ -0,0 +1,151 @@
+$ curl https://quay.io/cnr/api/v1/packages?namespace=certified-operators >> packages.txt
+----
++
+Each package in the new `packages.txt` is an Operator that you could add to your
+disconnected catalog. From this list, you could either pull every Operator or a
+subset that you would like to expose to users.
+
+. **Pull Operator content.**
++
+For a given Operator in the package list, you must pull its content. This
+example uses the `openshift-template-service-broker` Operator:
+
+.. Retrieve the digest:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages/redhat-operators/openshifttemplateservicebroker/3.0.0
+----
+
+.. From that JSON, take the digest and use it to pull the gzipped archive:
++
+----
+$ curl -XGET https://quay.io/cnr/api/v1/packages/redhat-operators/openshifttemplateservicebroker/blobs/sha256/9bddc82c0fee05b243d01c6e2f0110c413eb9165cd7bb8adc25f1e88997854c9 -o openshifttemplateservicebroker.tar.gz
+----
+
+.. To pull the information out, you must untar the archive into a directory with
+all the other Operators that you want. For example, to untar to a directory
+called `manifests/`:
++
+----
+$ tar fczP openshifttemplateservicebroker.tar.gz manifests/
+----
+
+.. In this directory, you must now handle two different types of bundles:
+
+... If you see a single file called `bundle.yaml`, you must break `data.clusterServiceVersion` (each file in the list), `data.Package`, and `data.customResourceDefinition` (each file in the list) into their own files:
++
+----
+manifests/
+└── openshifttemplateservicebroker
+      ├── clusterserviceversion.yaml
+      ├── customresourcedefinition.yaml
+      ├── package.yaml
+└── etcd-<version>
+  └─ <CSVs_and_CRDs_and_a_package_file>
+----
+
+... If you see files already in the structure described in the previous step, then you do not have to do anything.
+
+. **Create an Operator catalog image.**
+
+.. Save the following to a Dockerfile, for example named `example-registry.Dockerfile`:
++
+[source,go]
+----
+FROM registry.redhat.io/openshift4/ose-operator-registry:4.2 AS builder
+
+COPY manifests manifests
+
+RUN ./bin/initializer -o ./bundles.db
+
+FROM scratch
+
+COPY --from=builder /build/bundles.db /bundles.db
+COPY --from=builder /build/bin/registry-server /registry-server
+COPY --from=builder /bin/grpc_health_probe /bin/grpc_health_probe
+
+EXPOSE 50051
+
+ENTRYPOINT ["/registry-server"]
+
+CMD ["--database", "bundles.db"]
+----
+
+.. Use the `podman` command to create and tag the container image from the
+Dockerfile:
++
+----
+$ sudo podman build -f example-registry.Dockerfile \
+    -t <internal_registry>/<namespace>/example-registry:latest <1>
+----
+<1> Tag the image for the internal registry of the disconnected {product-title}
+cluster.
+
+.. Push the image to the internal registry of the disconnected {product-title}
+cluster:
++
+----
+$ sudo podman push <internal_registry>/<namespace>/example-registry:latest
+----
+
+. Create a CatalogSource pointing to the newly created image:
```

Comment:
How do we apply this file? 
`oc create -f file.yaml`, `oc apply .` Do we need a command here?

Reply:
👍 we should add this